numpyro/primitives.py

```diff
@@ -206,6 +206,49 @@ def deterministic(name, value):
     return msg['value']
 
 
+def _inspect():
+    """
+    EXPERIMENTAL Inspect the Pyro stack.
+
+    .. warning:: The format of the returned message may change at any time and
+        does not guarantee backwards compatibility.
+
+    :returns: A message with mask effects applied.
+    :rtype: dict
+    """
+    # NB: this is different from Pyro that in Pyro, all effects applied.
+    # Here, we only apply mask effect handler.
+    msg = {
+        "type": "inspect",
+        "fn": lambda: True,
+        "args": (),
+        "kwargs": {},
+        "value": None,
+        "mask": None,
```

Comment:
I can't anticipate the future usage of `_inspect`, so I have simplified it a bit.